### PR TITLE
feat(runtime-logger): Expose integration logger on runtime

### LINKED ISSUE #119: [pkg/modular-station]: Provide runtime `AstroIntegrationLogger`
Request by @Adammatthiesen

### PR DESCRIPTION
**Pull request by Fryuni**
- Closes #119

### CODE REVIEW

**Review by Adammatthiesen — APPROVED** [review]
Woot!

File: packages/runtime-logger/src/buildLoggerPlugin.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+			return null;
+		},
+		load(id) {
+			if (!id.startsWith(RESOLVED_MODULE_PREFIX)) return;
+			const loggerName = id.slice(RESOLVED_MODULE_PREFIX.length);
+			const logger = loggers.get(loggerName)!;
+			if (logger === undefined) return;
+
+			return `
+import { baseLogger } from '${INTERNAL_MODULE}';
```
Comment (florian-lefebvre) [inline packages/runtime-logger/src/buildLoggerPlugin.ts:31]: Would it be safer to use `JSON.stringify` each time?

File: packages/runtime-logger/src/devLoggerPlugin.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+			}
+			return null;
+		},
+		load(id) {
+			if (!id.startsWith(RESOLVED_MODULE_PREFIX)) return;
+			const loggerName = id.slice(RESOLVED_MODULE_PREFIX.length);
+			if (!loggers.has(loggerName)) return;
+
+			return `
+console.log(Symbol.for('${pluginName}'));
```
Comment (florian-lefebvre) [inline packages/runtime-logger/src/devLoggerPlugin.ts:29]: Forgotten?

File: packages/runtime-logger/src/internalPlugin.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	resolveId(id) {
+		if (id === INTERNAL_MODULE) {
+			return RESOLVED_INTERNAL_MODULE;
+		}
+	},
+	load(id) {
+		if (id !== RESOLVED_INTERNAL_MODULE) return;
+
+		return `
+const consoleLogDestination = {
```
Comment (florian-lefebvre) [inline packages/runtime-logger/src/internalPlugin.ts:17]: Maybe worth putting this in a stub file?

File: packages/runtime-logger/tsup.config.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { defineConfig } from 'tsup';
+
+export default defineConfig({
+	entry: ['src/index.ts'],
+	format: ['esm'],
+	target: 'node18',
+	bundle: true,
+	dts: {
+		banner: '/// <reference path="../virtual.d.ts" />\n',
```
Comment (florian-lefebvre) [inline packages/runtime-logger/tsup.config.ts:9]: I'm curious, why is this needed?

File: packages/runtime-logger/virtual.d.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+declare module '@it-astro:logger:*' {
```
Comment (florian-lefebvre) [inline packages/runtime-logger/virtual.d.ts:1]: Could it be useful to use `addDts` instead of the wildcard for users?

File: packages/runtime-logger/src/buildLoggerPlugin.ts (hunk 1 above)
Comment (Fryuni) [reply packages/runtime-logger/src/buildLoggerPlugin.ts:31]: Since it is a constant we control I don't think there is much benefit.

File: packages/runtime-logger/src/devLoggerPlugin.ts (hunk 2 above)
Comment (Fryuni) [reply packages/runtime-logger/src/devLoggerPlugin.ts:29]: 😓

File: packages/runtime-logger/virtual.d.ts (hunk 5 above)
Comment (Fryuni) [reply packages/runtime-logger/virtual.d.ts:1]: I see `addDts` as a tool to provide final users, author of Astro _project_, with custom types based on their use. It doesn't really apply here since this is a tool for integration authors that provides a type for those authors.

The dts wouldn't be generated for them to use in the integration code

File: packages/runtime-logger/tsup.config.ts (hunk 4 above)
Comment (Fryuni) [reply packages/runtime-logger/tsup.config.ts:9]: With this the types from `virtual.d.ts` are loaded just by importing this library. You don't need to add a reference yourself.

Comment (florian-lefebvre) [reply packages/runtime-logger/tsup.config.ts:9]: But aren't you importing `virtual.d.ts` already?

File: packages/runtime-logger/virtual.d.ts (hunk 5 above)
Comment (florian-lefebvre) [reply packages/runtime-logger/virtual.d.ts:1]: ah yeah true!

File: packages/runtime-logger/src/internalPlugin.ts [hunk 6]
```diff
@@ -1,8 +1,11 @@
 import type { Plugin } from 'vite';
+import { readFile } from 'fs/promises';
 
 export const INTERNAL_MODULE = '@it-astro:logger-internal';
 const RESOLVED_INTERNAL_MODULE = '\x00@it-astro:logger-internal';
 
+const templatePath = new URL('../template/loggerStub.mjs', import.meta.url);
```
Comment (florian-lefebvre) [inline packages/runtime-logger/src/internalPlugin.ts:7]: Don't forget to add `template` to `files` in the package.json if it's not the case already!

Comment (Fryuni) [reply packages/runtime-logger/src/internalPlugin.ts:7]: Thank you!

File: packages/runtime-logger/tsup.config.ts (hunk 4 above)
Comment (florian-lefebvre) [reply packages/runtime-logger/tsup.config.ts:9]: my bad i thought you were importing it

**Review by florian-lefebvre — APPROVED** [review]
